src/lxc/confile.c

```diff
@@ -487,6 +492,62 @@ static int set_config_net_veth_pair(const char *key, const char *value,
 	return network_ifname(netdev->priv.veth_attr.pair, value, sizeof(netdev->priv.veth_attr.pair));
 }
 
+static int set_config_net_veth_vlan_id(const char *key, const char *value,
+				  struct lxc_conf *lxc_conf, void *data)
+{
+	int ret;
+	struct lxc_netdev *netdev = data;
+
+	if (lxc_config_value_empty(value))
+		return clr_config_net_veth_vlan_id(key, lxc_conf, data);
+
+	if (!netdev)
+		return -1;
+
+	ret = get_u16(&netdev->priv.veth_attr.vlan_id, value, 0);
+	if (ret < 0)
+		return -1;
+
+	return 0;
+}
+
+static int set_config_net_veth_vlan_tagged_id(const char *key, const char *value,
+				       struct lxc_conf *lxc_conf, void *data)
+{
+	__do_free unsigned short *vlan_id = NULL;
+	__do_free struct lxc_list *list = NULL;
+	int ret;
+	char *netmask, *slash;
+	struct lxc_netdev *netdev = data;
+
+	if (lxc_config_value_empty(value))
+		return clr_config_net_veth_vlan_tagged_id(key, lxc_conf, data);
+
+	if (!netdev)
+		return ret_set_errno(-1, EINVAL);
+
+	vlan_id = malloc(sizeof(*vlan_id));
+	if (!vlan_id)
+		return -1;
+
+	ret = get_u16(vlan_id, value, 0);
+	if (ret < 0)
+		return -1;
+
+	list = malloc(sizeof(*list));
+	if (!list)
+		return -1;
+
+	lxc_list_init(list);
+	list->elem = vlan_id;
```

Comment:
can even do:
```
list->elem = move_ptr(vlan_id);
```
and
```
lxc_list_add_tail(&netdev->priv.veth_attr.vlan_tagged_ids, move_ptr(list));
```
as a pattern going forward. :)

Reply:
@brauner thanks have applied.